opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -513,6 +520,27 @@ static int mca_btl_tcp_create(int if_kindex, const char* if_name)
         /* allow user to override/specify latency ranking */
         sprintf(param, "latency_%s:%d", if_name, i);
         mca_btl_tcp_param_register_uint(param, NULL, btl->super.btl_latency, OPAL_INFO_LVL_5, &btl->super.btl_latency);
+
+        /* Only attempt to auto-detect bandwidth and/or latency if it is 0.
+         *
+         * If detection fails to return anything other than 0, set a default
+         * bandwidth and latency.
+         */
+        if (0 == btl->super.btl_bandwidth) {
+            unsigned int speed = opal_ethtool_get_speed(if_name);
+            btl->super.btl_bandwidth = (speed == 0) ? MCA_BTL_TCP_BTL_BANDWIDTH : speed;
+            if (i > 0) {
+                btl->super.btl_bandwidth >>= 1;
```

Comment:
@jsquyres - I commented about this above (pasting here):

_The bandwidth and latency bit shifting when dealing with interfaces with multiple links looks a bit odd to me. For example, an interface capable of 100 Mbits/sec and with 2 links, will have the 1st link's btl_bandwidth == 100, the second == 50. Assuming more links, each subsequent link's btl_bandwidth would also be set to 50. Is this accurate? The same kind of calculation happens for btl_latency._

Since this logic was in the original code, and I don't have the historical context, I've included it here as well.

@bosilca - Hi George, this bandwidth/latency shifts appear to have come from you. Could you please comment? Thanks!